File: projects/website/app/search/results/response.ts
import {Entry} from "./entry";
import {NbResults} from "./NbResults";
import {Filter} from "./Filter";

export class Response {
    constructor(public entries: Array<Entry>,
                public nbResults: NbResults = new NbResults()){}

    public countResults(filter: Filter = Filter.all): number {
        return this.nbResults.get(filter);
    }
}

File: projects/scala/src/main/scala/ws/exploration/attributes/Response.scala
package ws.exploration.attributes

import ws.indices.response.NbResults

case class Response(entries: List[Entry],
                    nbResults: NbResults) {
  def getEntry(i: Int)
  : Option[Entry] = i >= 0 && i < entries.size match {
    case true => Some(entries(i))
    case false => None
  }
}
